### ISSUE TITLE
Order of `xsd:element` and `xsd:group` not respected

### ISSUE THREAD

**Issue by SimonCockx** [xmlet/XsdParser#53]
**Describe the bug**
When parsed, the order of `xsd:element`s and `xsd:group`s in a sequence are not respected.

E.g.,
```
    <xsd:sequence>
          <xsd:element name="elem1" type="xsd:decimal" />
          <xsd:group ref="myGroup" />
          <xsd:element name="elem2" type="Currency" />
    </xsd:sequence>
```
When running `sequence.getElements()` on the parsed result, this returns a list of three `NamedConcreteElement`s, but in the wrong order:
- element named `elem1`
- element named `elem2`
- group named `myGroup`

I assume it is due to a bug in the [following method](https://github.com/xmlet/XsdParser/blob/master/src/main/java/org/xmlet/xsdparser/xsdelements/XsdMultipleElements.java#L43) defined in the `XsdMultipleElements` class:
```
    public void replaceUnsolvedElements(NamedConcreteElement elementWrapper) {
        if (elementWrapper.getElement() instanceof XsdElement){
            super.replaceUnsolvedElements(elementWrapper);
        }

        if (elementWrapper.getElement() instanceof XsdGroup){
            elements.add(elementWrapper);

            this.elements.removeIf(element ->
               element instanceof UnsolvedReference && compareReference(elementWrapper, (UnsolvedReference) element)
            );
        }
    }
```
Here groups are added to the end of the list while resolving references.

**Expected behavior**
I would expect the order to be retained.

**Library Version**
`1.2.4`

**Comment by SimonCockx** [comment]
@lcduarte I've raised a PR where I remove the faulty method. Is there any good reason why `replaceUnsolvedElements` was overridden?

**Comment by SimonCockx** [comment]
FYI, when cloning master and running a `mvn clean install` the test `testSchemaGetMethods` is failing for me. It seems there is a broken test on master.

**Comment by lcduarte** [comment]
Hello @SimonCockx 

Thanks for using the library and for raising the issue. I've checked the PR and I have my doubts about removing the method entirely, I'll need to review the history of the changes to refresh my memory, but if it passes the tests its most likely correct.

Regarding the failing test, it's weird, I've ran the tests locally on Tuesday and everything was fine, but I had the same issue in the past cloning the repo.

I'll try to check this issue either tomorrow or saturday.
